app/scripts/metamask-controller.js

```diff
@@ -2339,6 +2355,40 @@ export default class MetamaskController extends EventEmitter {
       clearPendingConfirmations.bind(this),
     );
 
+    // RemoteFeatureFlagController has subscription for preferences changes
+    this.controllerMessenger.subscribe(
+      'PreferencesController:stateChange',
+      previousValueComparator((prevState, currState) => {
+        const { useExternalServices: prevUseExternalServices } = prevState;
+        const { useExternalServices: currUseExternalServices } = currState;
+        if (currUseExternalServices && !prevUseExternalServices) {
+          this.remoteFeatureFlagController.enable();
+          this.remoteFeatureFlagController.updateRemoteFeatureFlags();
+        } else if (!currUseExternalServices && prevUseExternalServices) {
+          this.remoteFeatureFlagController.disable();
+        }
+      }, this.preferencesController.state),
+    );
+
+    // Initialize RemoteFeatureFlagController
+    this.remoteFeatureFlagController = new RemoteFeatureFlagController({
+      messenger: this.controllerMessenger.getRestricted({
+        name: 'RemoteFeatureFlagController',
+        allowedActions: ['PreferencesController:getState'],
+        allowedEvents: ['PreferencesController:stateChange'],
```

Comment:
The controller does not use these actions or events. See https://github.com/MetaMask/core/blob/e2cdac920d5fbf2a97172ba444dc239aca4fb359/packages/remote-feature-flag-controller/src/remote-feature-flag-controller.ts#L47

```suggestion
        allowedActions: [],
        allowedEvents: [],
```

Reply:
thanks ! addressed in e52c104c502a6b4a1e54a306b901e8cf70d2c5ba